base/observer_list_threadsafe.cc

```diff
@@ -6,15 +6,49 @@
 #include "base/compiler_specific.h"
 #include "third_party/abseil-cpp/absl/base/attributes.h"
 
+#if defined(STARBOARD)
+#include "starboard/once.h"
+#endif
+
 namespace base {
 namespace internal {
 
+#if defined(STARBOARD)
+
+namespace {
+
+ABSL_CONST_INIT SbOnceControl s_once_flag = SB_ONCE_INITIALIZER;
+ABSL_CONST_INIT SbThreadLocalKey s_thread_local_key = kSbThreadLocalKeyInvalid;
+
+void InitThreadLocalKey() {
+  s_thread_local_key = SbThreadCreateLocalKey(NULL);
+  DCHECK(SbThreadIsValidLocalKey(s_thread_local_key));
+}
+
+}
+
+void ObserverListThreadSafeBase::EnsureThreadLocalKeyInited() {
+  SbOnce(&s_once_flag, InitThreadLocalKey);
+  DCHECK(SbThreadIsValidLocalKey(s_thread_local_key));
+}
+
+const SbThreadLocalKey ObserverListThreadSafeBase::GetThreadLocalKey() {
+  return s_thread_local_key;
+}
+
+#else
 ABSL_CONST_INIT thread_local const ObserverListThreadSafeBase::
```

Comment:
Please file a feature request with platform/Starboard team for adding support of C++11 `thread_local` storage class specifier to the evergreen and modular builds, referencing this large and quite scary commit necessary to adapt. Future updates to latest upstream will be a lot easier with that support.

Reply:
Added a comment to b/310718608